doc/reference/reference_lua/box_space/create_index.rst

```diff
@@ -113,321 +124,404 @@ On this page:
+    box.space.tester:create_index('I2',{parts={{'x', 'scalar'}}})
+    box.space.tester:create_index('I3',{parts={{'x','scalar'},{'y','integer'}}})
+    box.space.tester:create_index('I4',{parts={{1,'scalar'}}})
+    box.space.tester:create_index('I5',{parts={{1,'scalar'},{2,'integer'}}})
+    box.space.tester:create_index('I6',{parts={1}})
+    box.space.tester:create_index('I7',{parts={1,2}})
+    box.space.tester:create_index('I8',{parts={'x'}})
+    box.space.tester:create_index('I9',{parts={'x','y'}})
+    box.space.tester:create_index('I10',{parts={{'x'}}})
+    box.space.tester:create_index('I11',{parts={{'x'},{'y'}}})
+
+.. _box_space-path:
+
+--------------------------------------------------------------------------------
+Using the path option for map fields (JSON-indexes)
+--------------------------------------------------------------------------------
+
+To create an index for a field that is a map (a path string and a scalar value),
+specify the path string during index_create, that is,
+:code:`parts={` :samp:`{field-number},'{data-type}',path = '{path-name}'` :code:`}`.
+The index type must be ``'tree'`` or ``'hash'`` and the field's contents
+must always be maps with the same path.
+
+**Example 1 -- The simplest use of path:**
+
+..  code-block:: lua
+
+    -- Result will be - - [{'age': 44}]
+    box.schema.space.create('T')
+    box.space.T:create_index('I',{parts={{field = 1, type = 'scalar', path = 'age'}}})
+    box.space.T:insert{{age=44}}
+    box.space.T:select(44)
+
+**Example 2 -- path plus format() plus JSON syntax to add clarity:**
+
+..  code-block:: lua
+
+    -- Result will be: - [1, {'FIO': {'surname': 'Xi', 'firstname': 'Ahmed'}}]
+    s = box.schema.space.create('T')
+    format = {{'id', 'unsigned'}, {'data', 'map'}}
+    s:format(format)
+    parts = {{'data.FIO["firstname"]', 'str'}, {'data.FIO["surname"]', 'str'}}
+    i = s:create_index('info', {parts = parts})
+    s:insert({1, {FIO={firstname='Ahmed', surname='Xi'}}})
+
+**Note re storage engine:** vinyl supports only the TREE index type, and vinyl
+secondary indexes must be created before tuples are inserted.
+
+.. _box_space-path_multikey:
+
+--------------------------------------------------------------------------------
+Using the path option with [*]
+--------------------------------------------------------------------------------
+
+The string in a path option can contain '[*]' which is called
+an array index placeholder. Indexes defined with this are useful
+for JSON documents that all have the same structure.
+
+For example, when creating an index on field#2 for a string document
+that will start with ``{'data': [{'name': '...'}, {'name': '...'}]``,
+the parts section in the create_index request could look like:
+``parts = {{field = 2, type = 'str', path = 'data[*].name'}}``.
+Then tuples containing names can be retrieved quickly with
+``index_object:select({key-value})``.
+
+In fact a single field can have multiple keys, as in this example
```

Comment:
1. In fact, a single field (comma)
2. The sentence is too complex. Try breaking it into two or three separate sentences.